### PR TITLE
New header on project page

### PR DESCRIPTION
**Pull request by denektenina**
# Description
Link to Jira issue: [https://sikt.atlassian.net/browse/NP-47426](https://sikt.atlassian.net/browse/NP-47426)

Add new header on project page to make it look different than the wizard on results-page:

image

I am trying to clean up placement of files a little bit when I encounter them, especially reusable components, so I moved a couple of them. Also made a new one out of existing code, since we need this icon-header more than one place now.

The strategy is: components-folder has everything reusable that is used in the whole app, while the pages-folders has components that are used for only those pages. Ish.

Also, I had to make a new PageHeader-component because the existing one didn't accept ReactComponents as children, and I didn't dare to change the existing one as it was used a lot of places. So now we have two 🥳

# Checklist

Ensure that the changes are aligned with the expectations of PO/designer before marking the PR as ready for review.

Also ensure that the following criterias are met:

- [x] The changes are working as expected
- [x] The changes are tested OK for different screen sizes
- [x] The changes are tested OK for a11y
- [x] Interactive elements have data-testids
- [x] I have done a QA of my own changes

_Note: some of these criterias may not always be relevant, and can simply be marked as completed._

### CODE REVIEW

File: src/components/page_layout_components/PageHeader.tsx
Comment (elkebab) [inline src/components/page_layout_components/PageHeader.tsx:1]: Kunne hatt dette og den gamle samlet i en `PageHeader.tsx` også med noe sånt:
```js
const PageHeader = (children) => 
  typeof children === "string" ? <StringPageHeader /> : <ReactNodePageHeader />
```

File: src/pages/project/project_wizard/ProjectForm.tsx [hunk 1]
```diff
@@ -84,7 +85,10 @@ export const ProjectForm = ({ project }: ProjectFormProps) => {
       <SkipLink href="#form">{t('common.skip_to_schema')}</SkipLink>
       <Formik initialValues={project} validationSchema={basicProjectValidationSchema} onSubmit={submitProjectForm}>
         <Form noValidate>
-          <PageHeader variant="h1">{project.title}</PageHeader>
+          <PageHeader>
+            <ProjectIconHeader projectStatus={projectWithId ? projectWithId.status : undefined} />
+            <Typography variant={'h1'}>{project.title}</Typography>
```
Comment (elkebab) [inline src/pages/project/project_wizard/ProjectForm.tsx:90]: Ser vi bruker `TruncatableTypography` i den gamle `PageHeader`, så kan være det hadde vært lurt her også, for å unngå at den tar veldig mye plass ved lange prosjektnavn. Trolig mest relevant på mobil, men vet ikke hvor lange navn vi reelt kan forvente da..

**Review by elkebab — DISMISSED** [review]
Nice!👍

File: src/pages/project/project_wizard/ProjectForm.tsx (hunk 1 above)
Comment (denektenina) [reply src/pages/project/project_wizard/ProjectForm.tsx:90]: Godt poeng!

**Review by elkebab — DISMISSED** [review]

**Review by elkebab — APPROVED** [review]

**Pull request merged**